From fe6570203f0eec5c9aabdc1943c1bd51bd5b5a8b Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Wed, 30 Oct 2019 11:19:35 +0000
Subject: [PATCH 1/7] inet: start dhcps always

This patch enables the dhcp server by default. 3rd party platforms
(for instance RDK-based) need this, as the dhcp server is independent of
OpenSync there.
---
 src/lib/inet/src/inet_base.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/lib/inet/src/inet_base.c b/src/lib/inet/src/inet_base.c
index a0fbda8..94a8e47 100644
--- a/src/lib/inet/src/inet_base.c
+++ b/src/lib/inet/src/inet_base.c
@@ -386,7 +386,7 @@ bool inet_base_init(inet_base_t *self, const char *ifname)
                     inet_unit_s(INET_BASE_NETWORK,
                             inet_unit_s(INET_BASE_SCHEME_NONE, NULL),
                             inet_unit(INET_BASE_SCHEME_STATIC,
-                                    inet_unit(INET_BASE_DHCP_SERVER, NULL),
+                                    inet_unit_s(INET_BASE_DHCP_SERVER, NULL),
                                     inet_unit(INET_BASE_DNS, NULL),
                                     NULL),
                             inet_unit(INET_BASE_SCHEME_DHCP, NULL),
@@ -1697,8 +1697,6 @@ bool inet_base_dhcp_server_commit(inet_base_t *self, bool start)
     if (self->in_dhcps != NULL) osn_dhcp_server_del(self->in_dhcps);
     self->in_dhcps = NULL;
 
-    if (!start) return true;
-
     /*
      * Create new DHCPv4 object
      */
-- 
2.25.1

